File: binding/src/svg/libqgraphicssvgitem.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qgraphicssvgitem.html
pub const qgraphicssvgitem = struct {

    /// New constructs a new QGraphicsSvgItem object.
    pub fn New() QtC.QGraphicsSvgItem {
        return qtc.QGraphicsSvgItem_new();
    }


    /// New2 constructs a new QGraphicsSvgItem object.
    pub fn New2(fileName: []const u8) QtC.QGraphicsSvgItem {
        const fileName_str = qtc.libqt_string{
    .len = fileName.len,
    .data = fileName.ptr,
};
return qtc.QGraphicsSvgItem_new2(fileName_str);
    }


    /// New3 constructs a new QGraphicsSvgItem object.
    pub fn New3(parentItem: ?*anyopaque) QtC.QGraphicsSvgItem {
        return qtc.QGraphicsSvgItem_new3(@ptrCast(parentItem));
    }


    /// New4 constructs a new QGraphicsSvgItem object.
    pub fn New4(fileName: []const u8, parentItem: ?*anyopaque) QtC.QGraphicsSvgItem {
        const fileName_str = qtc.libqt_string{
    .len = fileName.len,
    .data = fileName.ptr,
};
return qtc.QGraphicsSvgItem_new4(fileName_str, @ptrCast(parentItem));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QGraphicsSvgItem_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qgraphicssvgitem.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn SetSharedRenderer(self: ?*anyopaque, renderer: ?*anyopaque) void {
        qtc.QGraphicsSvgItem_SetSharedRenderer(@ptrCast(self), @ptrCast(renderer));
    }

    pub fn Renderer(self: ?*anyopaque, ) QtC.QSvgRenderer {
        return qtc.QGraphicsSvgItem_Renderer(@ptrCast(self));
    }

    pub fn SetElementId(self: ?*anyopaque, id: []const u8) void {
        const id_str = qtc.libqt_string{
    .len = id.len,
    .data = id.ptr,
};
qtc.QGraphicsSvgItem_SetElementId(@ptrCast(self), id_str);
    }

    pub fn ElementId(self: ?*anyopaque, , allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QGraphicsSvgItem_ElementId(@ptrCast(self));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qgraphicssvgitem.ElementId: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn SetCachingEnabled(self: ?*anyopaque, cachingEnabled: bool) void {
        qtc.QGraphicsSvgItem_SetCachingEnabled(@ptrCast(self), cachingEnabled);
    }

    pub fn IsCachingEnabled(self: ?*anyopaque, ) bool {
        return qtc.QGraphicsSvgItem_IsCachingEnabled(@ptrCast(self));
    }

    pub fn SetMaximumCacheSize(self: ?*anyopaque, size: ?*anyopaque) void {
        qtc.QGraphicsSvgItem_SetMaximumCacheSize(@ptrCast(self), @ptrCast(size));
    }

    pub fn MaximumCacheSize(self: ?*anyopaque, ) QtC.QSize {
        return qtc.QGraphicsSvgItem_MaximumCacheSize(@ptrCast(self));
    }

    pub fn BoundingRect(self: ?*anyopaque, ) QtC.QRectF {
        return qtc.QGraphicsSvgItem_BoundingRect(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnBoundingRect(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) QtC.QRectF) void {
        qtc.QGraphicsSvgItem_OnBoundingRect(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseBoundingRect(self: ?*anyopaque, ) QtC.QRectF {
        return qtc.QGraphicsSvgItem_QBaseBoundingRect(@ptrCast(self));
    }

    pub fn Paint(self: ?*anyopaque, painter: ?*anyopaque, option: ?*anyopaque, widget: ?*anyopaque) void {
        qtc.QGraphicsSvgItem_Paint(@ptrCast(self), @ptrCast(painter), @ptrCast(option), @ptrCast(widget));
    }

    /// Allows for overriding the related default method
    pub fn OnPaint(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, ?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QGraphicsSvgItem_OnPaint(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBasePaint(self: ?*anyopaque, painter: ?*anyopaque, option: ?*anyopaque, widget: ?*anyopaque) void {
        qtc.QGraphicsSvgItem_QBasePaint(@ptrCast(self), @ptrCast(painter), @ptrCast(option), @ptrCast(widget));
    }

    pub fn Type(self: ?*anyopaque, ) i32 {
        return qtc.QGraphicsSvgItem_Type(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnType(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) i32) void {
        qtc.QGraphicsSvgItem_OnType(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseType(self: ?*anyopaque, ) i32 {
        return qtc.QGraphicsSvgItem_QBaseType(@ptrCast(self));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QGraphicsSvgItem_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qgraphicssvgitem.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QGraphicsSvgItem_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qgraphicssvgitem.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QGraphicsSvgItem_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/graphicssvgitem.html#types
pub const enums = struct {
};

File: binding/src/svg/libqgraphicssvgitem.cpp
#include <QGraphicsItem>
#include <QGraphicsObject>
#include <QGraphicsSvgItem>
#include <QPainter>
#include <QRectF>
#include <QSize>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <type_traits>
#include <QStyleOptionGraphicsItem>
#include <QSvgRenderer>
#include <QWidget>
#include <qgraphicssvgitem.h>
#include "libqgraphicssvgitem.h"
#include "libqgraphicssvgitem.hxx"

QGraphicsSvgItem* QGraphicsSvgItem_new() {
	 return new VirtualQGraphicsSvgItem();
}

QGraphicsSvgItem* QGraphicsSvgItem_new2(const libqt_string fileName) {
	 return new VirtualQGraphicsSvgItem(QString::fromUtf8(fileName.data, fileName.len));
}

QGraphicsSvgItem* QGraphicsSvgItem_new3(QGraphicsItem* parentItem) {
	 return new VirtualQGraphicsSvgItem(parentItem);
}

QGraphicsSvgItem* QGraphicsSvgItem_new4(const libqt_string fileName, QGraphicsItem* parentItem) {
	 return new VirtualQGraphicsSvgItem(QString::fromUtf8(fileName.data, fileName.len), parentItem);
}

libqt_string QGraphicsSvgItem_Tr(const char* s) {
	QString _ret = QGraphicsSvgItem::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QGraphicsSvgItem_SetSharedRenderer(QGraphicsSvgItem* self, QSvgRenderer* renderer) {
	self->setSharedRenderer(renderer);
}

QSvgRenderer* QGraphicsSvgItem_Renderer(const QGraphicsSvgItem* self) {
	return self->renderer();
}

void QGraphicsSvgItem_SetElementId(QGraphicsSvgItem* self, const libqt_string id) {
	self->setElementId(QString::fromUtf8(id.data, id.len));
}

libqt_string QGraphicsSvgItem_ElementId(const QGraphicsSvgItem* self) {
	QString _ret = self->elementId();
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QGraphicsSvgItem_SetCachingEnabled(QGraphicsSvgItem* self, bool cachingEnabled) {
	self->setCachingEnabled(cachingEnabled);
}

bool QGraphicsSvgItem_IsCachingEnabled(const QGraphicsSvgItem* self) {
	return self->isCachingEnabled();
}

void QGraphicsSvgItem_SetMaximumCacheSize(QGraphicsSvgItem* self, const QSize* size) {
	self->setMaximumCacheSize(*size);
}

QSize* QGraphicsSvgItem_MaximumCacheSize(const QGraphicsSvgItem* self) {
	return new QSize(self->maximumCacheSize());
}

QRectF* QGraphicsSvgItem_BoundingRect(const QGraphicsSvgItem* self) {
	return new QRectF(self->boundingRect());
}

void QGraphicsSvgItem_Paint(QGraphicsSvgItem* self, QPainter* painter, const QStyleOptionGraphicsItem* option, QWidget* widget) {
	self->paint(painter, option, widget);
}

int QGraphicsSvgItem_Type(const QGraphicsSvgItem* self) {
	return self->type();
}

libqt_string QGraphicsSvgItem_Tr2(const char* s, const char* c) {
	QString _ret = QGraphicsSvgItem::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QGraphicsSvgItem_Tr3(const char* s, const char* c, int n) {
	QString _ret = QGraphicsSvgItem::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

// Base class handler implementation
QRectF* QGraphicsSvgItem_QBaseBoundingRect(const QGraphicsSvgItem* self) {
	auto* vqgraphicssvgitem = dynamic_cast<const VirtualQGraphicsSvgItem*>(self);
	if (vqgraphicssvgitem && vqgraphicssvgitem->isVirtualQGraphicsSvgItem) {
vqgraphicssvgitem->setQGraphicsSvgItem_BoundingRect_IsBase(true);
	return new QRectF(vqgraphicssvgitem->boundingRect());
}
}

// Auxiliary method to allow providing re-implementation
void QGraphicsSvgItem_OnBoundingRect(const QGraphicsSvgItem* self, intptr_t slot) {
	auto* vqgraphicssvgitem = dynamic_cast<const VirtualQGraphicsSvgItem*>(self);
	if (vqgraphicssvgitem && vqgraphicssvgitem->isVirtualQGraphicsSvgItem) {
vqgraphicssvgitem->setQGraphicsSvgItem_BoundingRect_Callback(reinterpret_cast<VirtualQGraphicsSvgItem::QGraphicsSvgItem_BoundingRect_Callback>(slot));
}
}

// Base class handler implementation
void QGraphicsSvgItem_QBasePaint(QGraphicsSvgItem* self, QPainter* painter, const QStyleOptionGraphicsItem* option, QWidget* widget) {
	auto* vqgraphicssvgitem = dynamic_cast<VirtualQGraphicsSvgItem*>(self);
	if (vqgraphicssvgitem && vqgraphicssvgitem->isVirtualQGraphicsSvgItem) {
vqgraphicssvgitem->setQGraphicsSvgItem_Paint_IsBase(true);
	vqgraphicssvgitem->paint(painter, option, widget);
}
}

// Auxiliary method to allow providing re-implementation
void QGraphicsSvgItem_OnPaint(QGraphicsSvgItem* self, intptr_t slot) {
	auto* vqgraphicssvgitem = dynamic_cast<VirtualQGraphicsSvgItem*>(self);
	if (vqgraphicssvgitem && vqgraphicssvgitem->isVirtualQGraphicsSvgItem) {
vqgraphicssvgitem->setQGraphicsSvgItem_Paint_Callback(reinterpret_cast<VirtualQGraphicsSvgItem::QGraphicsSvgItem_Paint_Callback>(slot));
}
}

// Base class handler implementation
int QGraphicsSvgItem_QBaseType(const QGraphicsSvgItem* self) {
	auto* vqgraphicssvgitem = dynamic_cast<const VirtualQGraphicsSvgItem*>(self);
	if (vqgraphicssvgitem && vqgraphicssvgitem->isVirtualQGraphicsSvgItem) {
vqgraphicssvgitem->setQGraphicsSvgItem_Type_IsBase(true);
	return vqgraphicssvgitem->type();
}
}

// Auxiliary method to allow providing re-implementation
void QGraphicsSvgItem_OnType(const QGraphicsSvgItem* self, intptr_t slot) {
	auto* vqgraphicssvgitem = dynamic_cast<const VirtualQGraphicsSvgItem*>(self);
	if (vqgraphicssvgitem && vqgraphicssvgitem->isVirtualQGraphicsSvgItem) {
vqgraphicssvgitem->setQGraphicsSvgItem_Type_Callback(reinterpret_cast<VirtualQGraphicsSvgItem::QGraphicsSvgItem_Type_Callback>(slot));
}
}

void QGraphicsSvgItem_Delete(QGraphicsSvgItem* self) {
    delete self;
}

